[voldiemort] iP

voldiemort opened the pull request:
# ~~Duke~~ Murphy Chatbot
## Changes made:
- renamed to Murphy (after Murphy's Law)
- added support for different task types
- date parsing and deletion of tasks
- many more (find out for yourself)

## How to get started:
1. Download the .jar file from the latest release [here](https://github.com/voldiemort/ip/releases)
2. cd to the directory containing the .jar file in your terminal
3. Run `java -jar murphy.jar`

## Upcoming features
- [x] search tasks by description
- [ ] add GUI
- [ ] more OOP

> "Don't trust every quote you see on the Internet." -Gandhi, 1994 💯 

Lastly a random snippet of code 
```java
public static void main(String[] args){
        new Murphy("./data/murphy.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

voldiemort committed:
renamed chatbot

voldiemort committed:
added greet and exit

voldiemort committed:
added echo

voldiemort committed:
added list and adding items to list

voldiemort committed:
added marking of tasks

voldiemort committed:
added unmarking

voldiemort committed:
added todo, deadline and event classes and adding them to list

voldiemort committed:
updated test case for I/O redirection testing

voldiemort committed:
added error messages for incorrect command usage

voldiemort committed:
changed tasks from array to arraylist

voldiemort committed:
added deleting of tasks

voldiemort committed:
program creates save file if not exist

voldiemort committed:
Add load save from file

voldiemort committed:
Add write save on program exit

voldiemort committed:
Merge branch 'branch-Level-7'

voldiemort committed:
Add date parsing for deadlines and events

voldiemort committed:
Merge branch 'branch-Level-8'

voldiemort committed:
Add Storage class

voldiemort committed:
Add tasklist implementation

voldiemort committed:
Integrate OOP classes

voldiemort committed:
Merge branch 'branch-A-MoreOOP'

voldiemort committed:
Move classes into packages

voldiemort committed:
Merge branch 'branch-A-Packages'

voldiemort committed:
Merge branch 'add-gradle-support'

voldiemort committed:
Modify app name in build.gradle

voldiemort committed:
Add DeadlineTest and TaskListTest

voldiemort committed:
Merge branch 'branch-A-JUnit'

voldiemort committed:
Add JavaDocs

voldiemort committed:
Modify some braces to Egyptian style

voldiemort committed:
Implement find function to filter by task description

voldiemort committed:
Merge branch 'branch-A-JavaDoc'

voldiemort committed:
Merge branch 'branch-A-CodingStandard'

voldiemort committed:
Merge branch 'branch-Level-9'

voldiemort committed:
Add checkstyle config files and modify code accordingly

RuijianLu commented on src/main/java/murphy/Murphy.java:
```diff
@@ -0,0 +1,161 @@
+        /*
+            String logo = " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
```
Should delete these commented lines since the ip grading doesn't allow lots of lines of commented code.

JiaYi-Gallium369 replied:
Yes, since they should already be captured in your git history.

RuijianLu commented on src/main/java/murphy/Murphy.java:
```diff
@@ -0,0 +1,161 @@
+        boolean isClosed = false;
+        while (!isClosed) {
+            String input = ui.getInput();
+            if (input.equals("bye")) {
```
Better divide these into another class `public class Parser` in package `Parser`.

JiaYi-Gallium369 replied:
Yes it's better to create a separate `Parser` class to encapsulate the commands as methods.

RuijianLu commented on src/main/java/murphy/task/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+import murphy.MurphyException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
`by` should have a more proper name, such as `ddlDate`.

JiaYi-Gallium369 replied:
the name `by` would be sufficient to tell the context.

RuijianLu commented on src/main/java/murphy/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+     * @throws MurphyException If description is empty.
+     */
+    public Task(String description) throws MurphyException {
+        String trimmed = description.trim();
```
`trimmed` can be `trimmedDescription`.

RuijianLu commented on src/main/java/murphy/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+        while (scanner.hasNext()) {
+            String nextTask = scanner.nextLine();
+            String[] taskDetails = nextTask.split("\\|");
+            if (Objects.equals(taskDetails[0], "T") && taskDetails.length == 3) {
+                Task task = new Todo(taskDetails[2], Boolean.parseBoolean(taskDetails[1]));
+                tasks.add(task);
+            } else if (Objects.equals(taskDetails[0], "D") && taskDetails.length == 4) {
+                Task task = new Deadline(taskDetails[2], Boolean.parseBoolean(taskDetails[1]),
+                        taskDetails[3]);
+                tasks.add(task);
+            } else if (Objects.equals(taskDetails[0], "E") && taskDetails.length == 5) {
+                Task task = new Event(taskDetails[2], Boolean.parseBoolean(taskDetails[1]),
+                        taskDetails[3], taskDetails[4]);
+                tasks.add(task);
```
These s would better be in the 'Storage` class.

RuijianLu commented on src/main/java/murphy/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * Finds tasks in the list which have a matching search term in the task description.
+     * @return The String representation of the filtered TaskList.
```
an empty line should be between method description and @ .

RuijianLu commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,43 @@
+____________________
+Got it. I've added this task:
+[T][ ] test todo
+Now you have 1 task(s) in the list.
+____________________
+Got it. I've added this task:
+[D][ ] homework (by: Sunday)
+Now you have 2 task(s) in the list.
+____________________
+Got it. I've added this task:
+[E][ ] dinner (from: 6pm to: 9pm)
+Now you have 3 task(s) in the list.
+____________________
+1. [T][ ] test todo
+2. [D][ ] homework (by: Sunday)
+3. [E][ ] dinner (from: 6pm to: 9pm)
+____________________
+Nice! I've marked this task as done:
+[D][X] homework (by: Sunday)
+____________________
+1. [T][ ] test todo
+2. [D][X] homework (by: Sunday)
+3. [E][ ] dinner (from: 6pm to: 9pm)
+____________________
+I've unmarked this task. Guess Murphy struck?
+[D][ ] homework (by: Sunday)
+____________________
+1. [T][ ] test todo
+2. [D][ ] homework (by: Sunday)
+3. [E][ ] dinner (from: 6pm to: 9pm)
+____________________
+Got it. I've deleted this task:
+[E][ ] dinner (from: 6pm to: 9pm)
+Now you have 2 task(s) in the list.
+____________________
+1. [T][ ] test todo
+2. [D][ ] homework (by: Sunday)
+____________________
+Bye. Hope to see you again soon!
+____________________
```
Just `gitignore` this.

RuijianLu reviewed (commented):
LGTM but better code structure can be avhieved.

JiaYi-Gallium369 commented on build.gradle:
```diff
@@ -0,0 +1,46 @@
+    archiveClassifier = null
+}
+
+run{
```
Add space before curly brace

JiaYi-Gallium369 commented on src/main/java/murphy/ui/Ui.java:
```diff
@@ -0,0 +1,36 @@
+
+    public void showText(String text) {
+        System.out.println(text);
+        System.out.println("____________________");
```
can create a constant for `"____________________"`

JiaYi-Gallium369 reviewed (commented):
Code looks good overall! Good job making use of OOP for Tasks

voldiemort committed:
Integrate GUI

voldiemort committed:
Merge branch 'branch-Level-10'

voldiemort committed:
Add some simple assertions

voldiemort committed:
Improve code quality

Code was slightly messy with some violations of SLAP and some
too long methods.

Code inside these methods were abstracted out into new methods.

voldiemort committed:
Merge pull request #2 from voldiemort/branch-A-Assertions

Add some simple assertions

voldiemort committed:
Merge branch 'master' into branch-A-CodeQuality

voldiemort committed:
Merge pull request #3 from voldiemort/branch-A-CodeQuality

Improve code quality

voldiemort committed:
Add view schedule feature to see tasks on a date

For Deadline tasks, the task is shown as long as the deadline is after
the date specified.

For Event tasks, the task is shown if the date specified is within range
of the event.

voldiemort committed:
Merge branch 'branch-B-ViewSchedules'

voldiemort committed:
Add user guide and screenshot
